connector/signaltometricsconnector/internal/trace/adjustedcount.go

```diff
@@ -0,0 +1,45 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package trace // import "github.com/elastic/opentelemetry-collector-components/connector/signaltometricsconnector/internal/trace"
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/sampling"
+)
+
+// CalculateAdjustedCount calculates the adjusted count which represents
+// the number of spans in the population that are represented by the
+// individually sampled span. If the span is not-sampled OR if a non-
+// probability sampler is used then adjusted count defaults to 1.
+// https://github.com/open-telemetry/oteps/blob/main/text/trace/0235-sampling-threshold-in-trace-state.md
+func CalculateAdjustedCount(tracestate string) int64 {
+	w3cTraceState, err := sampling.NewW3CTraceState(tracestate)
+	if err != nil {
+		return 1
+	}
+	otTraceState := w3cTraceState.OTelValue()
+	if otTraceState == nil {
+		return 1
+	}
+	if len(otTraceState.TValue()) == 0 {
+		// For non-probabilistic sampler OR always sampling threshold, default to 1
+		return 1
+	}
+	// TODO (lahsivjar): Handle fractional adjusted count. One way to do this
+	// would be to scale the values in the histograms for some precision.
```

Comment:
In the context of this being called from a span converter now, I think the approach to scale the histograms isn't really feasible.

I wonder if the adjusted count should return a floating-point number so that users can decide what to do with it. But that makes it harder on the user of this converter and it means that we'd probably need something like a `ProbabilisticRound()` converter at some point. Maybe it makes sense to pull forward the probabilistic rounding so that this returns an integer, which is easer to work with?

Reply:
Hmm good point on the return type of floating-point. I am not sure if I fully understand all the concerns.

> I wonder if the adjusted count should return a floating-point number so that users can decide what to do with it.

This sounds like a good idea to me, I didn't quite get why it would get harder for the user of this converter as they could do one of the following:

1. `Int(AdjustedCount())` - if the want int
2. `Double(AdustedCount())` - if they want float
3. `ProbabilisticRound(AdustedCount())` - if they want rouding

Reply:
> I didn't quite get why it would get harder for the user of this converter

It's just that they need to know how to deal with the floating-point number and they may not know what's be best strategy in which situation. Also, there are more things that can go wrong. For example, when just using `AdustedCount()` when an integer is expected. We'll also need to try to upstream both `AdjustedCount()` and `ProbabilisticRound()`.

Despite all the tradeoffs, it's probably still the best option.

Reply:
> It's just that they need to know how to deal with the floating-point number and they may not know what's be best strategy in which situation

Internally, we would have to cast the float64 to int to be able to use it as a count for histogram. So default usage of `AdustedCount()` should also work as expected.

Also, the [upstream package](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/47118f27b17fb275eddf2725a4a373939fbce2a1/pkg/sampling/oteltracestate.go#L182) we use returns `float64` which should make the behavior more consistent.